src/GameState.cpp

```diff
@@ -320,6 +320,48 @@ std::optional<std::optional<Color>> GameState::getOutcome() const {
     // TODO:
     // If we're in check, see if there's a way to get out. (If not, our opponent wins.)
     // If we're not in check, see if there's any move that doesn't put us in check. (If not, it's stalemate.)
+
+
+    // Check for checkmate if king is in check
+    if (this->isCheck(this->getTurn()))
+    {
+        int legalMoveFlag = 0;
+        for (const auto [square, piece] : pieceMap) {
+            // Iterating over the pieces of the same color and checking for legal moves
+            if (piece.color != this->turn) continue;
+
+            for (auto move : this->generateMoves(square)) {
+                if(isLegal(move))
```

Comment:
`wouldBeInCheck()` suffices here, as in the stalemate loop below. (The first thing `isLegal()` does is generate all the moves from the source square, so it'll duplicate work here.)